operator/v1/types_ingress.go

```diff
@@ -233,6 +233,97 @@ type LoadBalancerStrategy struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Scope LoadBalancerScope `json:"scope"`
+
+	// providerParameters holds desired load balancer information
+	// specific to the underlying infrastructure provider.
+	//
+	// If empty, defaults will be applied. See specific ProviderLoadBalancerParameters
+	// fields for details about their defaults.
+	//
+	// +optional
+	ProviderParameters ProviderLoadBalancerParameters `json:"providerParameters,omitempty"`
+}
+
+// ProviderLoadBalancerParameters holds desired load balancer information specific
+// to the underlying infrastructure provider.
+// +union
+type ProviderLoadBalancerParameters struct {
+	// type is the underlying infrastructure provider for the load balancer.
+	// The only allowed value is "AWS".
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +required
+	Type PlatformType `json:"type"`
```

Comment:
Just to be clear, `type` needs to be allowed to be empty on non-AWS platforms, and we may want to add additional allowed values to `PlatformType` in the future (because we may want to add parameters for additional platforms).  Do these definitions allow for this?  Probably a question for @deads2k again.

Reply:
@Miciah `subfields` (formerly `providerLoadBalancerParameters `) is optional, so I thought requiring `type` and only allowing AWS parameters would suffice. Let's see what @deads2k has to say.

Reply:
[L271](https://github.com/openshift/api/pull/672/files?file-filters%5B%5D=.go#diff-85d84174432a6d2df99d107129e012f8R271) adds other supported providers even though [L267](https://github.com/openshift/api/pull/672/files?file-filters%5B%5D=.go#diff-85d84174432a6d2df99d107129e012f8R267) ("AWS") is the only provider that exposes LB configuration parameters.